src/Controller/SantaController.php

```diff
@@ -224,49 +216,47 @@ public function sendSampleMessage(Request $request, string $application): Respon
         }
 
         $session = $request->getSession();
-        $availableUsers = $session->get('available-users', []);
-        $selectedUsers = $session->get('selected-users', []);
-        $message = $session->get('message', '');
-        $options = $session->get('options', []);
 
-        $form = $this->createForm(MessageType::class, ['message' => $message], [
-            'selected-users' => $selectedUsers,
+        /** @var Config $config * */
+        $config = $session->get('config');
+
+        $builder = $formFactory->createBuilder(MessageType::class, $config, [
+            'selected-users' => $config->getSelectedUsers(),
         ]);
-        $form->handleRequest($request);
 
-        $message = $form->getData()['message'] ?? '';
+        $application->configureMessageForm($builder);
 
-        $notes = [];
-        foreach ($form->getData() as $data => $note) {
-            if (str_contains($data, 'notes-') && $note) {
-                $notes[str_replace('notes-', '', $data)] = $note;
-            }
-        }
+        $form = $builder->getForm();
+
+        $form->handleRequest($request);
+
+        $notes = $config->getNotes();
 
-        $candidates = array_filter($notes ? array_keys($notes) : $selectedUsers, function ($id) use ($application) {
+        $candidates = array_filter($notes ? array_keys($notes) : $config->getSelectedUsers(), function ($id) use ($application) {
             return $application->getAdmin()->getIdentifier() !== $id;
         });
 
         $receiver = $candidates ? $candidates[array_rand($candidates)] : $application->getAdmin()->getIdentifier();
```

Comment:
Could you move the notes & candidates inside the block `$form->isSubmitted()`?

Reply:
👌